lib/sink.js

```diff
@@ -73,61 +71,95 @@ class WriteStream extends stream.PassThrough {
 }
 
 class ReadStream extends stream.PassThrough {
-    constructor(gcs, bucket, file, options) {
+    constructor(gcsFile, fileName, options) {
         super();
-        const gcsFile = bucket.file(file);
         const gcsStream = gcsFile.createReadStream(options);
 
         gcsStream.on('response', () => {
-            this.emit('file found', file);
+            this.emit('file found', fileName);
         });
 
         gcsStream.on('error', () => {
-            this.emit('file not found', file);
+            this.emit('file not found', fileName);
         });
 
         gcsStream.pipe(this);
     }
 }
 
 module.exports = class SinkGCS extends EventEmitter {
-    constructor(options, bucket, metadata = {}) {
+    constructor(options, bucketName, metadata = {}) {
         super();
         assert(
             options && isObject(options),
             '"options" object must be provided'
         );
-        assert(bucket && isString(bucket), '"bucket" string must be provided');
+        assert(
+            bucketName && isString(bucketName),
+            '"bucket" string must be provided'
+        );
 
         this.options = options;
+        this.bucketName = bucketName;
         this.name = 'asset-pipe-sink-gcs';
         this.gcs = gcsClient(options);
         this.bucket = null;
 
-        this.gcs.createBucket(bucket, metadata, (error, bucketObj) => {
-            if (error) {
-                this.bucket = this.gcs.bucket(bucket);
-                return this.emit(
+        this.ready = this.gcs
+            .createBucket(bucketName, metadata)
+            .then(bucketObj => {
+                this.bucket = bucketObj;
+                this.emit(
+                    'storage info',
+                    `Bucket "${bucketName}" did not exist. Created bucket`
+                );
+                return this.bucket;
+            })
+            .catch(error => {
+                this.bucket = this.gcs.bucket(bucketName);
+                this.emit(
                     'storage info',
-                    `Bucket "${bucket}" exist. Using bucket`,
+                    `Bucket "${bucketName}" exist. Using bucket`,
                     error
                 );
-            }
-            this.bucket = bucketObj;
-            this.emit(
-                'storage info',
-                `Bucket "${bucket}" did not exist. Created bucket`
-            );
-        });
+                return this.bucket;
+            });
+    }
+
+    _getGcsFile(fileName) {
+        return this.gcs.bucket(this.bucketName).file(fileName);
+    }
+
+    async get(fileName) {
+        const data = await this._getGcsFile(fileName).download();
```

Comment:
should we be try/catching this and wrapping the error or are we happy with whatever error GCS might return reaching the consumer?

Reply:
Unsure, if something is wrong - what kind of errors should we suppress?

Reply:
Not sure what I was thinking, maybe some kind of wrapping like:

`'sink.get' was unable to retrieve file '${fileName}' from storage. Additionally got error: ${error}`

or some such. Might be a bit overkill?